code/datums/lazy_template.dm

```diff
@@ -81,12 +90,25 @@
 		for(var/turf/turf as anything in block(bottom_left, top_right))
 			loaded_turfs += turf
 			loaded_areas |= get_area(turf)
-			for(var/thing in turf.get_all_contents())
-				// atoms can actually be in the contents of two or more turfs based on its icon/bound size
-				// see https://www.byond.com/docs/ref/index.html#/atom/var/contents
+
+			// atoms can actually be in the contents of two or more turfs based on its icon/bound size
+			// see https://www.byond.com/docs/ref/index.html#/atom/var/contents
+			for(var/thing in (turf.get_all_contents() - turf))
+				// check for specific type handling
+				for(var/load_type in type_to_list)
+					if(!istype(thing, load_type))
+						continue
+					type_to_list[load_type] += thing
+					break
+
 				loaded_atom_movables |= thing
 
-	SSatoms.InitializeAtoms(loaded_atom_movables + loaded_turfs + loaded_areas)
+	SSatoms.InitializeAtoms(loaded_areas)
+	SSatoms.InitializeAtoms(loaded_atom_movables)
+	SSatoms.InitializeAtoms(loaded_turfs)
```

Comment:
why separate?

Reply:
I had issues locally where it kept initializing stuff in the wrong order, it needs to be A -> AM -> T for undertile stuff to work correctly

Reply:
not sure if it's just my local Byond being stupid tho

Reply:
What it's doing before should be fine? InitializeAtoms isn't magic, so what you describe doesn't really make sense

Reply:
for the record I could see something like late init needing to happen before another init, but that'd be a bug tbh